### PR TITLE
Fix optimized remove in comparison logic

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: syncd/syncd_applyview.cpp [hunk 1]
```diff
@@ -1369,6 +1410,11 @@ class AsicView
                  * If operation is found then we need to insert this op after
                  * current iterator and forward iterator.
                  *
+                 * But here is a catch here, if taht last operation was REMOVE,
```
Comment (lguohan) [inline syncd/syncd_applyview.cpp]: >taht [](start = 48, length = 4)

that

File: syncd/syncd_applyview.cpp [hunk 2]
```diff
@@ -1330,6 +1350,27 @@ class AsicView
...
+                 *
+                 * this logic could result with wrong order:
+                 * - remove D
+                 * - remove C (will break reference count on C, since used in B)
+                 *   since D is considered last OP releasing C reference
+                 * - create A
+                 * - set object B with A
+                 *
+                 * This is fixed by checking if last operation to release reference was REMOVE
+                 */
```
Comment (lguohan) [inline syncd/syncd_applyview.cpp]: If i understand correctly, with your fix, the optimized order is as follows?

remove D
create A
set object B with A
remove C

File: syncd/syncd_applyview.cpp [hunk 3]
```diff
@@ -1381,6 +1427,23 @@ class AsicView
                             sai_serialize_object_id(op.vid).c_str());
                 }
 
+                if (itr->isRemove)
+                {
+                    SWSS_LOG_INFO("previous operation to release reference %s was also REMOVE, we push current REMOVE op at the list end",
+                            sai_serialize_object_id(mit->first).c_str());
+
+                    /*
+                     * If previous operation was REMOVE (it could be pused to
```
Comment (lguohan) [inline syncd/syncd_applyview.cpp]: pushed

File: meta/sai_meta.cpp [hunk 4]
```diff
@@ -232,6 +232,41 @@ static std::unordered_map<sai_object_id_t,int32_t> ObjectReferences;
...
+
+            case SAI_OBJECT_TYPE_LAG:
+            case SAI_OBJECT_TYPE_NEXT_HOP:
+            case SAI_OBJECT_TYPE_NEXT_HOP_GROUP:
+            case SAI_OBJECT_TYPE_NEXT_HOP_GROUP_MEMBER:
+            case SAI_OBJECT_TYPE_ROUTE_ENTRY:
+            case SAI_OBJECT_TYPE_ROUTER_INTERFACE:
+                break;
+
+                // skip object we have no intrest into
```
[the remark this answers is not in the document]
Comment (jleveque) [reply meta/sai_meta.cpp]: Also change "into" to "in"

File: syncd/syncd_applyview.cpp [hunk 5]
```diff
@@ -13,6 +13,13 @@
  * parameters to every function.
  */
 
+/**
+ * When set to true extra logging will be added for tracking references.  This
+ * is useful for duugging, but for production operations this will produce too
```
Comment (stcheng) [inline syncd/syncd_applyview.cpp]: debugging

**Pull request merged**